src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1038,14 +982,41 @@ void Graph::PullOutputData(BlobMap &out) {
 
         auto srcPrec = actualDesc.getPrecision();
         auto dstPrec = expectedDesc.getPrecision();
-
-        if (!getConfig().isLegacyApi && srcPrec == dstPrec && ext_blob->byteSize() != intr_blob.getSize())
-            IE_THROW() << "Output blob byte size is not equal network output byte size (" << ext_blob->byteSize()
+        if (!getConfig().isLegacyApi && srcPrec == dstPrec && ext_blob.get_byte_size() != intr_blob.getSize())
+            IE_THROW() << "Output blob byte size is not equal network output byte size (" << ext_blob.get_byte_size()
                        << "!=" << intr_blob.getSize() << ").";
 
-        void *ext_blob_ptr = ext_blob->buffer();
+        void *ext_blob_ptr = ext_blob.data();
         void *intr_blob_ptr = intr_blob.getData();
 
+        // If output precision has been changed comparing to original model's output, it must be copied to aux tensor
+        if (out_precision_changed[name]) {
+            auto it = aux_tensors.find(name);
+            if (it == aux_tensors.end()) {
+                OPENVINO_THROW("Output precision has been changed, but cannot find its aux tensor.");
+            }
+
+            auto& aux_tensor = it->second;
+            aux_tensor.set_shape(out[name].get_shape());
+            ext_blob_ptr = aux_tensor.data();
+            if ((intr_blob_ptr == nullptr) || (ext_blob_ptr == nullptr)) {
+                OPENVINO_THROW("Get tensor has no allocated memory");
+            }
+
+            cpu_convert(intr_blob_ptr,
+                        ext_blob_ptr,
+                        srcPrec,
+                        InferenceEngine::details::convertPrecision(aux_tensor.get_element_type()),
+                        intr_blob.getDescWithType<BlockedMemoryDesc>()->getPaddedElementsCount());
+
+            if (actualDesc.getBlockingDesc() != expectedDesc.getBlockingDesc() && !isScalarOutput) {
+                // Reorder if shape or stride has been changed, but it will impact performance
+                // Add assert here to capture this performance issue.
+                OPENVINO_ASSERT("Precision changes and reorder occurred at the same time.");
+            }
+            continue;
+        }
```

Comment:
If we move it out of graph level, it will need copy twice: 1)internal mem --> output tensor  2) output tensor --> aux tensor.
While if put it here, we only need copy once: internal mem --> aux tensor, and aux tensor will return to user directly.

Reply:
Why can't we put the `aux_tensor` to the `out` map instead of the original tensor?

Reply:
`m_out` map is not new created memory, it points to the same memory of m_tensors

1) `m_tensors` is managed by infer_request, which can be found by port object (compiled model), and `m_out` is taken out from `m_tensors` (output port's tensor), they point to the same memory, we can look `m_out` as the reference of output port's  tensor in `m_tensors`. It will be convenient to use `m_out `to represent output tensors of `m_tensors.`

2) `m_aux_tensors` is new created tensor to used to solve precision change issue between original model and compiled model, its precision is same to original model's input/output port, and it always is read/written by user if it exists.

![image](https://github.com/openvinotoolkit/openvino/assets/31196718/788e0916-9080-4b5f-9c82-f6859d73bcf3)